src/traces/scattergl/convert.js

```diff
@@ -279,18 +281,24 @@ proto.updateFast = function(options) {
         yy = y[i];
 
         // check for isNaN is faster but doesn't skip over nulls
-        if(!isNumeric(yy)) continue;
-        if(!isNumeric(xx) && !(xx instanceof Date)) continue;
+        fastType = isNumeric(xx) || xx instanceof Date;
 
-        idToIndex[pId++] = i;
+        if(isNumeric(yy) && (fastType || Lib.isDateTime(xx))) {
 
-        positions[ptr++] = xx;
-        positions[ptr++] = yy;
+            if(!fastType) {
+                xx = Lib.dateTime2ms(xx);
+            }
```

Comment:
@etpinard @alexcjohnson as you see I ended up *not* skimping on tests, i.e. not just checking the first value, as we already had a check with relatively fast tests (number or `Date`). It's only when this fails is when the slower path is taken. If you think I can optimize away the `isDateTime` part, i.e. just assume that once we hit one of those, the rest of them will be like it.